crates/sp-domains/src/lib.rs

```diff
@@ -1473,6 +1473,16 @@ impl<Balance> OnChainRewards<Balance> for () {
     fn on_chain_rewards(_chain_id: ChainId, _reward: Balance) {}
 }
 
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub enum OperatorRewardSource<Number> {
+    Bundle {
+        at_block_number: Number,
+    },
+    XDMProtocolFees,
+    #[cfg(any(feature = "std", feature = "runtime-benchmarks"))]
```

Comment:
It seems like this dummy is only used in tests and benchmarks, so I’m not sure we want it enabled for all std code?

```suggestion
    #[cfg(any(test, feature = "runtime-benchmarks"))]
```

Reply:
I considered using `test` but interestingly it doesn't compile...

Reply:
That’s concerning, because I can only see `Dummy` used inside `test` and `feature = "runtime-benchmarks"` in this PR. So `#[cfg(any(test, feature = "runtime-benchmarks"))]` should be enough to make it compile.

What’s the exact compile error?
Is that something you can check, or would you like me to?

Reply:
Oh, I see why this happens. `subspace-test-service` is compiled as a production binary, not a `#![cfg(test)]` binary.

We could fix this by adding a new feature to `sp-domains`, and making the test crates depend on it.

Reply:
>subspace-test-service is compiled as a production binary, not a #![cfg(test)] binary.

The unit tests of `pallet-domains` also use `OperatorRewardSource::Dummy`. And there are other `#[cfg(any(feature = "std", feature = "runtime-benchmarks"))]` exist due to this reason.

To fix them we need to add a new feature to `sp-domains` and add a new dependency with this feature under the `[dev-dependencies]` in all the crates that use it. Let's leave it as is for this PR, I will open an issue for this and handle it later.